### PR TITLE
SWIFT-1054 Unified test runner updates for versioned API

### PR DESCRIPTION
**Pull request by kmahar**
Makes some updates to the runner that are prerequisites for decoding and running the versioned API tests:

* Support for decoding more types of operations we hadn't added before
* Support for a new type of test requirement, required server parameters

### CODE REVIEW

File: Sources/MongoSwift/MongoCollection+BulkWrite.swift [hunk 1]
```diff
@@ -325,6 +325,18 @@ public struct BulkWriteOptions: Codable {
...
+
+    private enum CodingKeys: String, CodingKey {
+        case bypassDocumentValidation, ordered, writeConcern
+    }
+
+    // A manual implementation is required to enforce a default value for ordered without making it optional.
+    public init(from decoder: Decoder) throws {
+        let container = try decoder.container(keyedBy: CodingKeys.self)
+        self.bypassDocumentValidation = try container.decodeIfPresent(Bool.self, forKey: .bypassDocumentValidation)
+        self.ordered = try container.decodeIfPresent(Bool.self, forKey: .ordered) ?? true
```
Comment (patrickfreed) [inline Sources/MongoSwift/MongoCollection+BulkWrite.swift:337]: if we provide a default value like `public var ordered = true` does the synthesized implementation not use that when ordered is missing?

**Review by patrickfreed — COMMENTED** [review]
All looks good, just have one minor question about a `Codable` conformance.

**Comment by patrickfreed** [comment]
Oh, also looks like sourcery needs to be run.

File: Sources/MongoSwift/MongoCollection+BulkWrite.swift (hunk 1 above)
Comment (kmahar) [reply Sources/MongoSwift/MongoCollection+BulkWrite.swift:337]: I had the same thought. unfortunately no, that default value does not get propagated to the Codable conformance.

**Comment by kmahar** [comment]
whoops, updated linuxmain.

**Review by patrickfreed — APPROVED** [review]

**Review by bynn — APPROVED** [review]
everything looks good

**Pull request merged**